### PR TITLE
Food-App Rails Group project

### PR DESCRIPTION
**Pull request by cosywasswa**
👇 In this project we completed the following tasks.

- [ ] Followed the layout of the wireframes given in the requirements.
- [ ] Login page and registration page created with devise
- [ ] Food list created
     -displays a list of food added by the logged-in user as in the wireframe (display also quantity of a givenfood).
      -leads to a form that allows users to add new food.

- [ ] Recipes list:
       -display a list of recipes created by the logged-in user as in the wireframe.
      -lead to recipe details.
      -If the user is the owner of the recipe, should allow the user to delete it.

- [ ] Public recipe list:
      -display a list of all public recipes ordered by newest as in the wireframe.
      -lead to recipe details.

- [ ] Recipe details:
     -display a toggle button that allows for a recipe to be made public or private.
     -If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
     -If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- [ ] General shopping list view:
     -shows the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes 
             with the general food list of that user).
      -counts the total food items and total price of the missing food.

- [ ] No N+1 queries happening
- [ ] Created a navigation menu that allows users to open all of the pages 
- [ ] Wrote unit and integration tests
- [ ] Updated README file with project information
- [ ] Fixed linter errors.

### CODE REVIEW

File: app/views/devise/confirmations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend confirmation instructions" %>
```
Comment (mukhammadrizooff) [inline app/views/devise/confirmations/new.html.erb:12]: - [x] [REQUIRED] Good job so far! Kindly on my side I didn't have recieve `confirmation code` to my email, please fix it and check again it'll work or not 🙏🏼 

image

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @TEAM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Team work ✔️ 
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- All Specs passed ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
- [x] [REQUIRED FOR NEXT CR] Dear CR, I checked all codebase and specs, they looks well and passed, kindly check layout as `Live` because due to the fact that I could not go to the next pages from the `login` page 🙏🏼 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/confirmations/new.html.erb (hunk 1 above)
Comment (AJ193) [reply app/views/devise/confirmations/new.html.erb:12]: Dear Next Reviewer The solution to this problem is basically explained in this video so kindly watch this and then you would be able to go through that issue.
Thanks

https://github.com/cosywasswa/Foods-App/assets/119049613/bda714e7-05d4-4c7c-96ec-42825fef05dd

File: app/views/recipes/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <div class="field">
+      <%= f.label :description, class: "form-label" %><br />
+        <%= f.text_area :description, autofocus: true, autocomplete: "description" , class: "mb-3 form-control" %>
+    </div>
+    <div class="mb-3 form-check">
+      <%= f.radio_button :public, true %>
+        <%= f.label 'Public' , :public=> 'True' %>
+
+          <%= f.radio_button :public, false %>
+            <%= f.label 'Private' , :public=> 'False' %>
```
Comment (Lucas-Masaba) [inline app/views/recipes/new.html.erb:26]: - [ ] When creating a private recipe, I can still view it on the public list and yet I am not supposed to.  For example, I made Krabby Patty public and Secret formula private as seen in the example bellow but I can see them both on the public recipes tab. Please fix that.
image

File: app/views/recipes/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        </h4>
+        <h4>Steps go there...</h4>
+      </div>
+      <div class="table-cont">
+    
+      <%= form_for @recipe, url: public_toggle_user_recipe_path(@recipe), method: :patch, remote: true, authenticity_token: true, html: { class: 'toggle-form' } do |f| %>
+        <div class="slider-container">
+        <span class="switch-label">Public</span>
+        <label class="switch">
+          <%= f.check_box :public, { class: "toggle-switch" }, 0, 1 %>
```
Comment (Lucas-Masaba) [inline app/views/recipes/show.html.erb:21]: - [ ] When I change a recipe to either public or private and refresh the page or go to another page, it goes back to what it originally was. Please make sure that when I change the recipe status to public/private it stays that way

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @cosywasswa && @AJ193   👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost-there

 ## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Most features implemented :gear: 
- Good communication with code reviewer :microphone: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @team  , 

Great job so far!  

May I highlight the following: 
- [x]  Well done implementing the required changes by the previews TSE's👏  ✔️ 
- [x] Your code is clean and well documented✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**